### PR TITLE
Code quality review using Rector

### PR DESCRIPTION
**Pull request by dafeder**
New run of rector, updating config and removing some file and rule skips. Attempted to bump PHP target to 8.1 but created too much delta, will do that in another pass. 

Also made some manual fixes of things that jumped out at my local linter.

### CODE REVIEW

File: modules/datastore/src/Drush.php [hunk 1]
```diff
@@ -186,13 +182,23 @@ public function list($options = [
    * Private.
    */
   private function createRow($uuid, $item) {
+    // Using deprecation helper.
     return [
       'uuid' => $uuid,
       'fileName' => $item->fileName,
       'fileFetcherStatus' => $item->fileFetcherStatus,
-      'fileFetcherBytes' => \format_size($item->fileFetcherBytes) . " ($item->fileFetcherPercentDone%)",
+      'fileFetcherBytes' => DeprecationHelper::backwardsCompatibleCall(
```
Comment (dafeder) [inline modules/datastore/src/Drush.php:190]: A little on the fence about whether to use this Deprecation method, just jump to ByteSizeMarkup and break Drupal 10.1 (which is EOL), or skip the change completely for now.

File: modules/datastore/src/DataDictionary/AlterTableQueryBase.php [hunk 2]
```diff
@@ -14,43 +14,31 @@ abstract class AlterTableQueryBase implements AlterTableQueryInterface {
...
    * Query table.
-   *
-   * @var string
    */
   protected string $table;
 
   /**
    * Query fields.
-   *
-   * @var array
```
Comment (paul-m) [inline modules/datastore/src/DataDictionary/AlterTableQueryBase.php:46]: It's fine that we remove `@var` here, but we also don't have inline documentation of how to define fields or indexes. We should add that in the `__construct()` docblock, but that can be a follow-up.

File: modules/datastore/src/PostImportResult.php [hunk 3]
```diff
@@ -43,29 +43,21 @@ class PostImportResult {
...
    * The PostImport service.
-   *
-   * @var \Drupal\datastore\Service\PostImport
    */
   protected PostImport $postImport;
 
   /**
    * The PostImportResult.
-   *
-   * @var array
```
Comment (paul-m) [inline modules/datastore/src/PostImportResult.php:68]: `$postImportResult` and `$resourceMapper` seem to be unused in this class. Let's do a followup to remove them.

File: modules/datastore/tests/src/Kernel/DatastoreServiceEventsTest.php [hunk 4]
```diff
@@ -40,15 +40,13 @@ class DatastoreServiceEventsTest extends KernelTestBase implements EventSubscrib
 
   /**
    * Store the events we receive.
-   *
-   * @var array
```
Comment (paul-m) [inline modules/datastore/tests/src/Kernel/DatastoreServiceEventsTest.php:44]: should be `@var \Drupal\datastore\Events\DatastoreEventBase[]`

File: modules/datastore/tests/src/Kernel/Service/ImportServiceEventsTest.php [hunk 5]
```diff
@@ -37,15 +37,13 @@ class ImportServiceEventsTest extends KernelTestBase implements EventSubscriberI
 
   /**
    * Store the events we receive.
-   *
-   * @var array
```
Comment (paul-m) [inline modules/datastore/tests/src/Kernel/Service/ImportServiceEventsTest.php:41]: Should be `@var \Drupal\datastore\Events\DatastoreImportedEvent[]`

**Review by paul-m — CHANGES_REQUESTED** [review]

File: modules/datastore/src/DataDictionary/AlterTableQueryBase.php (hunk 2 above)
Comment (dafeder) [reply modules/datastore/src/DataDictionary/AlterTableQueryBase.php:46]: OK will do

Comment (dafeder) [reply modules/datastore/src/DataDictionary/AlterTableQueryBase.php:46]: See #4369

File: modules/datastore/src/PostImportResult.php (hunk 3 above)
Comment (dafeder) [reply modules/datastore/src/PostImportResult.php:68]: Addressing this in #4370

**Review by paul-m — APPROVED** [review]

**Pull request merged**